skimage/measure/_boundary.py

```diff
@@ -0,0 +1,160 @@
+import numpy as np
+
+"""
+Code copied from https://github.com/machine-shop/deepwings/blob/master/deepwings/method_features_extraction/image_processing.py#L185
+"""
+
+
+def _moore_neighborhood(current, backtrack):  # y, x
+    """Returns clockwise list of pixels from the moore neighborhood of current
+    pixel:
+    The first element is the coordinates of the backtrack pixel.
+    The following elements are the coordinates of the neighboring pixels in
+    clockwise order.
+    Parameters
+    ----------
+    current ([y, x]): Coordinates of the current pixel
+    backtrack ([y, x]): Coordinates of the backtrack pixel
+    Returns
+    -------
+    List of coordinates of the moore neighborood pixels, or 0 if the backtrack
+    pixel is not a current pixel neighbor
+    """
+
+    operations = np.array(
+        [[-1, 0], [-1, 1], [0, 1], [1, 1], [1, 0], [1, -1], [0, -1], [-1, -1]]
+    )
+    neighbors = (current + operations).astype(int)
+
+    for i, point in enumerate(neighbors):
+        if np.all(point == backtrack):
+            # we return the sorted neighborhood
+            return np.concatenate((neighbors[i:], neighbors[:i]))
+    else:
+        raise RuntimeError("The backtrack is not on the neighborhood")
+
+
+def trace_boundary(coords):
+    """Coordinates of the region's boundary. The region must not have isolated
+    points.
+    
+    Parameters
+    ----------
+    coords : obj
+        Obtained with skimage.measure.regionprops()
+        
+    Returns
+    -------
+    boundary : 2D array
+        List of coordinates of pixels in the boundary
+        The first element is the most upper left pixel of the region.
+        The following coordinates are in clockwise order.
+    """
+
+    def starting_point():
+        """
+        Returns where we will start the search for the boundary. Current it
+        returns (max_x, max_y). That is, the basic example:
+
+            [0, 0, 0, 0, 0, 0],
+            [0, 1, 1, 1, 1, 0],
+            [0, 1, 1, 1, 1, 0],
+            [0, 0, 1, 1, 1, 0],
+            [0, 0, 0, 1, 0, 0],
+
+        Will start in the (4, 4) point because it has a region limit at
+        (1, 4) and another at (4, 3). On a t-shaped example, where the region
+        has borders in very far away points, say (1, 20) and (20, 1), I fear
+        that this method will begin its search at (20, 20), a point very
+        far from the boundary
+        """
+        return np.amax(coords, axis=0)
```

Comment:
Is it worth it defining the `starting_point` function for a one line call?

Reply:
No not really. And it's something that should be solved differently in Cython where there is no need to check every element. We don't really need the position of a global maximum but only the edge of the object.